src/cryptography/x509/base.py

```diff
@@ -183,6 +187,51 @@ def public_bytes(self, encoding):
         Serializes the certificate to PEM or DER format.
         """
 
+    def verify(self, intermediates, trusted_roots, cert_verif_cb):
+        """
+        Find chains of trust from `self` to any certificate in `trusted_roots`,
+        using certificates from `intermediates`.
+        :param intermediates: list of Certificate objects to build chains from
+        :param trusted_roots: list of Certificate objects which are trusted
+        :param cert_verif_cb: ignored for now. Will take (chain: List(Certificate), position: int)
+        :return: Iterable of lists of Certificate objects. Each list is a trust
+            chain starting with `self` and ending with one of `trusted_roots`.
+        """
+
+def default_cert_verif_cb(chain, i):
+    """
+    Reasonable minimal default for the callback, applied to each certificate in the
+    trust chain.
+    For the chain to be valid, the callback must return True for all its certificates.
+    This example checks validity time, CA bits and path length.
+
+    :param chain: list of Certificate objects such as for i in range(len(chain) - 1),
+                  issuer(chain[i]) == subject(chain[i+1].
+                  The last (chain[-1]) certificate can be self-signed or not.
+    :param i: index of the checked certificate in the chain
+    :return: True if the certificate passes checks, False otherwise
+    """
+    cert = chain[i]
+
+    # Check that CA bit is set for all certs but the leaf
+    if i > 0:
+        bc = cert.extensions.get_extension_for_oid(
+            ExtensionOID.BASIC_CONSTRAINTS
+        )
+        if bc is not None:
+            if not bc.value.ca:
+                return False
+
+            # Check that path length constraint, if present, is fulfilled
+            if bc.value.path_length is not None and bc.value.path_length > i:
+                return False
+
+    # Check validity period
+    now = datetime.datetime.now()
```

Comment:
```suggestion
    now = datetime.datetime.utcnow()
```
I believe the `not_valid_before` and `not_valid_after` fields are expected to be encoded in UTC (or GMT to be very precise, but gmt is a timezone, utc is a time standard, and in this context they are interchanged). Therefore I would think we should compare against a UTC datetime.

Reply:
Thank you for the correction!